TRex2-lib/src/Packets/RulePkt.cc

```diff
@@ -46,33 +46,33 @@ RulePkt* RulePkt::copy() {
 //#endif
 
 RulePkt::RulePkt(bool resetCount) {
-  if (resetCount)
+  if (resetCount) {
     lastId = 0;
+  }
   ruleId = lastId++;
   ceTemplate = NULL;
 }
 
 RulePkt::~RulePkt() {
-  for (map<int, Predicate>::iterator it = predicates.begin();
-       it != predicates.end(); ++it) {
-    delete it->second.constraints;
+  for (const auto& it : predicates) {
+    delete[] it.constraints;
```

Comment:
BUGFIX: I changed from `delete` to `delete[]`, because I guess we are trying to delete an array and not a single object, right?